test/rule-matches/widget-not-inline-matches.js

```diff
@@ -1,124 +1,197 @@
         '<svg role="button" tabindex="0" id="target"></svg>'
       );
-      var node = vNode.actualNode;
+      const node = vNode.actualNode;
       assert.isFalse(rule.matches(node, vNode));
     });
 
-    it('returns false descendants of SVG elements', function () {
-      var vNode = queryFixture(
+    it('returns false descendants of SVG elements', () => {
+      const vNode = queryFixture(
         '<svg><a href="#" id="target">' +
           '  <text y="15">link</text></a>' +
           '</svg>'
       );
-      var node = vNode.actualNode;
+      const node = vNode.actualNode;
       assert.isFalse(rule.matches(node, vNode));
     });
   });
+
+  describe('nested widget', () => {
+    describe('that are is in the tab order', () => {
+      it('is true when the target is in the tab order', () => {
+        const vNode = queryFixture(`
+          <span role="link" tabindex="0">
+            Link text
+            <a href="#" id="target">Nested link</a>
+          </span>`);
+        const node = vNode.actualNode;
+        assert.isTrue(rule.matches(node, vNode));
+      });
+
+      it('is false when the target is not in the tab order', () => {
+        const vNode = queryFixture(`
+          <span role="link" tabindex="0">
+            Link text
+            <a href="#" id="target" tabindex="-1">Nested link</a>
+          </span>`);
+        const node = vNode.actualNode;
+        assert.isFalse(rule.matches(node, vNode));
+      });
+
+      describe('within a shadow DOM tree', () => {
+        it('is true when the target is in the tab order', () => {
+          const vNode = queryShadowFixture(
+            `<div role="link" tabindex="0"><span id="shadow"></span></div>`,
+            `<a href="#" id="target">Nested link</a>`
+          );
+          const node = vNode.actualNode;
+          assert.isTrue(rule.matches(node, vNode));
+        });
+
+        it('is false when the target is not in the tab order', () => {
+          const vNode = queryShadowFixture(
+            `<div role="link" tabindex="0"><span id="shadow"></span></div>`,
+            `<a href="#" id="target" tabindex="-1">Nested link</a>`
+          );
+          const node = vNode.actualNode;
+          assert.isFalse(rule.matches(node, vNode));
+        });
+      });
+    });
+
+    describe('that is not in the tab order', () => {
+      it('is true when the target is in the tab order', () => {
+        const vNode = queryFixture(`
+          <span role="link" tabindex="-1">
+            Link text
+            <a href="#" id="target">Nested link</a>
+          </span>`);
+        const node = vNode.actualNode;
+        assert.isTrue(rule.matches(node, vNode));
+      });
+
+      it('is true when the target is not in the tab order', () => {
+        const vNode = queryFixture(`
+          <span role="link" tabindex="-1">
+            Link text
+            <a href="#" id="target" tabindex="-1">Nested link</a>
+          </span>`);
```

Comment:
What does this as an integration test produce? Would the child link report a target-size violation but the parent would pass?